src/v2/cookbook/vue-route-change-handling.md

```diff
@@ -0,0 +1,376 @@
+    }
+  });
+}, 300),
+```
+
+Only issue now is the `page` stays the same, so the logical way would be to tweak it a bit:
+
+```js
+searchUpdated: debounce(function (event) {  
+  this.$router.push({ 
+    query: {
+      ...this.$route.query,
+      page: 1, // Reset the page
+      filter: event.target.value,
+    }
+  });
+}, 300),
+```
+
+But what happens if the user changes the `filter` in the url directly? The answer is - whatever the `$route` watcher does. In this case we technically lost the page reset functionality, while it's arguable whether or not this is a valid case as the user can change the `page` into invalid value and it breaks all the same, it would still be preferable that by changing the `filter` query (maybe it happened programmatically with `$router.push/replace`) the component would deal with resetting `page` itself. While the latter would be ideal, it would require writing an advanced navigation interceptor to combine multiple component query requirements and patch the destination route to avoid multiple update triggers and history entries.
+
+<p class="tip">Note! There is no way to reset query values or know if they're invalid if the user entered the page with already invalid query parameters until the data is downloaded or known before hand.</p>
+
+```js
+$route(to, from) {
+  // Ommited for the sake of sanity...
+  // Because of the complexity only 2 fields introduce,
+  // you can easily fall into infinite redirect loop and
+  // just hang the users browser.
+}
+```
+
+Going back to the order it all happens - user changes search query, a code snippet updates the query with new search term, a `$route` watcher fires that resets the `page` parameter in similar way as `searchUpdated` except uses `$router.replace`, the watcher fires again to make sure everything is in order this time (and maybe again... and again... and so on if there's even a small mistake).
+
+So at this point you'd rather deal with reloading most of the data and be happy with your life, if only you weren't a developer and this wouldn't haunt you that is... and it just keeps piling on - what if the query parameter is a `tab` name, it should not reload anything on the page, just switch the active tab; what if I have index view, and the edit is a modal on top of the view with it's unique route (e.g. `order.edit`), why opening it reloads the list, and so on... these are the things that the `<router-view :key="$route.fullPath"/>` prevents you from achieving, that and the insane amounts of code and logic required to handle the `$route` watcher.
+
+## Partial solution
+While advanced solution could be invented, for most of the use cases a more linear approach is sufficient, so in spite of that, the current solution has limitations:
+
+ - if the user changes url query directly, it will not reset query values (without navigation interception this could cause multiple history entries added when values are being reset);
```

Comment:
Please capitalize the first word and put a period at the end of each sentence for each li :)